bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewHubHandler.java

```diff
@@ -110,25 +138,37 @@ public void dispose() {
         stopPoll();
     }
 
-    void pollNow() {
-        if (isInitialized()) {
-            schedulePoll();
+    private void schedulePoll() {
+        ScheduledFuture<?> future = this.pollFuture;
+        if (future != null && !future.isCancelled()) {
+            future.cancel(false);
         }
-    }
+        logger.debug("Scheduling poll for 5000ms out, then every {}ms", refreshInterval);
+        this.pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 5000, refreshInterval, TimeUnit.MILLISECONDS);
 
-    private void schedulePoll() {
-        if (pollFuture != null) {
-            pollFuture.cancel(false);
+        future = this.hardRefreshFuture;
+        if (future != null && !future.isCancelled()) {
+            future.cancel(false);
+        }
+        if (hardRefreshInterval > 0) {
+            logger.debug("Scheduling hard refresh every {}minutes", hardRefreshInterval);
+            this.hardRefreshFuture = scheduler.scheduleWithFixedDelay(this::requestRefreshShades, 1,
+                    hardRefreshInterval, TimeUnit.MINUTES);
         }
-        logger.debug("Scheduling poll for 500ms out, then every {} ms", refreshInterval);
-        pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 500, refreshInterval, TimeUnit.MILLISECONDS);
     }
 
     private synchronized void stopPoll() {
-        if (pollFuture != null && !pollFuture.isCancelled()) {
-            pollFuture.cancel(true);
-            pollFuture = null;
+        ScheduledFuture<?> future = this.pollFuture;
+        if (future != null && !future.isCancelled()) {
+            future.cancel(true);
```

Comment:
For further coding: No harm in canceling an already cancelled task. The `isCancelled()` can be removed safely.

Reply:
Removed.